mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/InMemoryMessageManagerTest.java

```diff
@@ -18,22 +18,55 @@
  ****************************************************************/
 package org.apache.james.mailbox.inmemory;
 
+import static org.apache.james.mailbox.fixture.MailboxFixture.ALICE;
+import static org.apache.james.mailbox.fixture.MailboxFixture.BOB;
+import static org.apache.james.mailbox.fixture.MailboxFixture.INBOX_ALICE;
+
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MailboxSessionUtil;
+import org.apache.james.mailbox.fixture.MailboxFixture;
 import org.apache.james.mailbox.inmemory.manager.InMemoryIntegrationResources;
-import org.apache.james.mailbox.store.AbstractMessageManagerTest;
+import org.apache.james.mailbox.store.MessageManagerContract;
 import org.apache.james.mailbox.store.MessageManagerTestSystem;
-import org.junit.Before;
+import org.junit.jupiter.api.BeforeEach;
 
-public class InMemoryMessageManagerTest extends AbstractMessageManagerTest {
+class InMemoryMessageManagerTest implements MessageManagerContract {
 
-    @Override
-    @Before
-    public void setUp() throws Exception {
-        super.setUp();
+    MailboxManager mailboxManager;
+    MailboxSession aliceSession;
+    MailboxSession bobSession;
```

Comment:
Idem, these fields (sessions) should be in the contract.

Reply:
Well the issue is that it cannot create a new mailbox in the second test if the sessions are being initialized in the contract... Anyway we decided after discussion to keep the abstract class and drop the interface conversion, as it will keep the code simple.